enarx-keep-sgx-shim/src/start.S

```diff
@@ -163,6 +196,28 @@ syscall:
 
     xor     %rcx,                   %rcx            # Clear %rcx
     zerof   %rcx                                    # Clear CPU flags
+
+    # Save stack pointer, allocate space, and set
+    # 64-byte alignment on stack for xsavec
+    mov     %rsp,                   %r8
```

Comment:
What is the purpose of this line? The `%r8` register should be passed directly to the outside world. But you're modifying it here.

Reply:
Oh, I see what you're trying to do. You can't store `%rsp` in `%r8` for reuse later in `.Lsyscall`.